modules/javafx.web/src/main/java/com/sun/webkit/network/HTTP2Loader.java

```diff
@@ -0,0 +1,648 @@
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.SequenceInputStream;
+import java.io.UnsupportedEncodingException;
+import java.lang.annotation.Native;
+import java.net.ConnectException;
+import java.net.CookieHandler;
+import java.net.HttpRetryException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.NoRouteToHostException;
+import java.net.SocketException;
+import java.net.SocketTimeoutException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.net.URLDecoder;
+import java.net.UnknownHostException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse.BodyHandler;
+import java.net.http.HttpResponse.BodySubscriber;
+import java.net.http.HttpResponse;
+import java.net.http.HttpTimeoutException;
+import java.nio.ByteBuffer;
+import java.security.AccessControlException;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Vector;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Flow;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.InflaterInputStream;
+import javax.net.ssl.SSLHandshakeException;
+import static com.sun.webkit.network.URLs.newURL;
+import static java.net.http.HttpClient.Redirect;
+import static java.net.http.HttpClient.Version;
+import static java.net.http.HttpResponse.BodyHandlers;
+import static java.net.http.HttpResponse.BodySubscribers;
+
+final class HTTP2Loader extends URLLoaderBase {
+
+    private static final PlatformLogger logger =
+            PlatformLogger.getLogger(URLLoader.class.getName());
+
+    private final WebPage webPage;
+    private final boolean asynchronous;
+    private String url;
+    private String method;
+    private final String headers;
+    private FormDataElement[] formDataElements;
+    private final long data;
+    private volatile boolean canceled = false;
+
+    private final CompletableFuture<Void> response;
+    // Use singleton instance of HttpClient to get the maximum benefits
+    private final static HttpClient HTTP_CLIENT = HttpClient.newBuilder()
+                   .version(Version.HTTP_2)  // this is the default
+                   .followRedirects(Redirect.NEVER) // WebCore handles redirection
+                   .connectTimeout(Duration.ofSeconds(30)) // FIXME: Add a property to control the timeout
+                   .cookieHandler(CookieHandler.getDefault())
+                   .build();
```

Comment:
This needs to be in a doPrivileged block. Otherwise WebView will fail when run with a security manager enabled. Here is the stack trace:

```
Exception in thread "JavaFX Application Thread" java.lang.ExceptionInInitializerError
	at javafx.web/com.sun.webkit.network.NetworkContext.fwkLoad(NetworkContext.java:161)
	at javafx.web/com.sun.webkit.WebPage.twkOpen(Native Method)
	at javafx.web/com.sun.webkit.WebPage.open(WebPage.java:1236)
	at javafx.web/javafx.scene.web.WebEngine.load(WebEngine.java:892)
	at hello.HelloWebView.start(HelloWebView.java:131)
	at javafx.graphics/com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$9(LauncherImpl.java:846)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runAndWait$12(PlatformImpl.java:455)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:428)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:389)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:427)
	at javafx.graphics/com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96)
	at javafx.graphics/com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at javafx.graphics/com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$11(GtkApplication.java:277)
	at java.base/java.lang.Thread.run(Thread.java:835)
Caused by: java.security.AccessControlException: access denied ("java.net.NetPermission" "getCookieHandler")
	at java.base/java.security.AccessControlContext.checkPermission(AccessControlContext.java:472)
	at java.base/java.security.AccessController.checkPermission(AccessController.java:1042)
	at java.base/java.lang.SecurityManager.checkPermission(SecurityManager.java:408)
	at java.base/java.net.CookieHandler.getDefault(CookieHandler.java:75)
	at javafx.web/com.sun.webkit.network.HTTP2Loader.<clinit>(HTTP2Loader.java:109)
	... 14 more
```

Reply:
The following should fix this:

```
            private final static HttpClient HTTP_CLIENT =
                    AccessController.doPrivileged((PrivilegedAction<HttpClient>) () -> HttpClient.newBuilder()
                       .version(Version.HTTP_2)  // this is the default
                       .followRedirects(Redirect.NEVER) // WebCore handles redirection
                       .connectTimeout(Duration.ofSeconds(30)) // FIXME: Add a property to control the timeout
                       .cookieHandler(CookieHandler.getDefault())
                       .build());
```